ui/src/hooks.js

```diff
@@ -10,3 +22,97 @@ export const useTypedSelector: <TSelected>(
   selector: (state: RootState) => TSelected,
   equalityFn?: (left: TSelected, right: TSelected) => boolean,
 ) => TSelected = useSelector;
+
+/**
+ * It retrieves the nodes data through react-queries
+ */
+export const useNodes = (): Nodes => {
+  const [nodes, setNodes] = useState([]);
+
+  const nodesQuery = useQuery(
+    'nodesNames',
+    () =>
+      coreV1.listNode().then((res) => {
+        if (res.response.statusCode === 200 && res.body?.items)
+          return res.body?.items;
+        return null;
+      }),
+    {
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      refetchInterval: REFRESH_METRICS_GRAPH,
+      refetchIntervalInBackground: true,
+    },
+  );
+
+  useEffect(() => {
+    if (!nodesQuery.isLoading && nodesQuery.isSuccess) {
+      setNodes(nodesQuery.data);
+    }
+  }, [nodesQuery]);
+
+  return nodes;
+};
+
+export const useNodeAddressesSelector = (nodes: Nodes): NodesAddresses => {
```

Comment:
```suggestion
export const useNodeAddressesSelector = (nodes: Nodes): Array<{internalIP: string, name: string}>=> {
```
IMO, I would be easier to read if we directly type here.